AssetManagementTestProject/TestCases/US307_EditUserTest.cs

```diff
@@ -0,0 +1,88 @@
+﻿using AssetManagementTestProject.DAO;
+using AssetManagementTestProject.PageObj;
+using AssetManagementTestProject.TestData;
+using AssetManagementTestProject.TestSetup;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+
+namespace AssetManagementTestProject.TestCases;
+
+[TestFixture]
+public class US307_EditUserTest : NUnitWebTestSetup
+{    
+    protected ManageUserPage? ManageUserPage; 
+    protected EditUserInfoPage? EditUserInfoPage;
+    protected DetailedUserInfoPage? DetailedUserInfoPage;
+    protected ViewUserDAO.ViewDetailedUser? DetailedUserInfoFromUI;
+    protected GetUserDAO.GetUserResponse? NewUserAfterEdit;
+    [Test]
+    public void TC01_AdminEditUserInfoSuccessfully()
+    {
+        LoginPage?.Login(Constant.ADMIN_USERNAME_HN, Constant.ADMIN_PASSWORD);
+        DriverBaseAction?.WaitToBeVisible(HomePage.HeaderHomePage);
+        DriverBaseAction?.Click(MenuBarLeft.BtnManageUserInMenu);
+        ManageUserPage = new ManageUserPage();
+        DriverBaseAction?.WaitToBeVisible(ManageUserPage.HeaderUserList);
+        ManageUserPage.InputSearch(NewUser.Data.StaffCode);
+        // Verify that the search user is displayed on top before editing
+        Asserter?.AssertEquals(ManageUserPage.ReturnStaffCodeTopListUser(), NewUser.Data.StaffCode);
+        DriverBaseAction?.Click(ManageUserPage.BtnEditUserAtTop);
+        EditUserInfoPage = new EditUserInfoPage();
+        EditUserInfoPage.EditUserInfo(EditUserData.EDITED_USER);
+        DriverBaseAction?.WaitToBeVisible(ManageUserPage.BtnEditUserAtTop);
+        // Screen shot to ensure edited user is at the top of the list
+        DriverBaseAction?.FindElementByXpath(ManageUserPage.FirstRowOfUserList);
+        DriverBaseAction?.Click(ManageUserPage.BtnViewTopUserDetailedInfo);
+        DetailedUserInfoPage = new DetailedUserInfoPage();
+        DriverBaseAction?.WaitToBeVisible(DetailedUserInfoPage.HeaderDetailedUser);
+        DetailedUserInfoPage detailedUserInfo = new DetailedUserInfoPage();
+        DetailedUserInfoFromUI = detailedUserInfo.ReturnDetailedUserInfoFromUI();
+        // Update new user info with GET request
+        NewUserAfterEdit = APIService?.ReturnSelectedUser(NewUser.Data.Id.ToString(), Token);
```

Comment:
This is incorrect.
If you are trying to edit a user, then you have to create the expected data by yourself.
IF you get data from API to be the expected data, THEN you're verifying the database with the database.
It's always true except the convert function from FrontEnd have some issues.

Reply:
Done